alf/algorithms/algorithm.py

```diff
@@ -1533,6 +1714,214 @@ def _update(self, experience, batch_info, weight):
+                # error: https://github.com/pytorch/pytorch/issues/59756
+                indices = alf.nest.utils.convert_device(
+                    torch.randperm(batch_size, device='cpu'))
+
+            for b in range(0, batch_size, mini_batch_size):
+                if update_counter_every_mini_batch:
+                    alf.summary.increment_global_counter()
+
+                is_last_mini_batch = (u == num_updates - 1
+                                      and b + mini_batch_size >= batch_size)
+                do_summary = (is_last_mini_batch
+                              or update_counter_every_mini_batch)
+                alf.summary.enable_summary(do_summary)
+
+                if indices is None:
+                    batch_indices = slice(b,
+                                          min(batch_size, b + mini_batch_size))
+                else:
+                    batch_indices = indices[b:min(batch_size, b +
+                                                  mini_batch_size)]
+
+                if experience is not None:
+                    batch = alf.nest.map_structure(lambda x: x[batch_indices],
+                                                   experience)
+                    if batch_info:
+                        binfo = alf.nest.map_structure(
+                            lambda x: x[batch_indices] if isinstance(
+                                x, torch.Tensor) else x, batch_info)
+                    else:
+                        binfo = None
+                    batch = _make_time_major(batch)
+                    batch = alf.nest.utils.convert_device(batch)
+                else:
+                    batch = None
+                    binfo = None
+
+                offline_batch = alf.nest.map_structure(
+                    lambda x: x[batch_indices], offline_experience)
+                if offline_batch_info:
+                    offline_binfo = alf.nest.map_structure(
+                        lambda x: x[batch_indices] if isinstance(
+                            x, torch.Tensor) else x, offline_batch_info)
+                else:
+                    offline_binfo = None
+                offline_batch = _make_time_major(offline_batch)
+                offline_batch = alf.nest.utils.convert_device(offline_batch)
+
+                (exp, train_info, loss_info, offline_exp, offline_train_info,
+                 offline_loss_info, params) = self._hybrid_update(
+                     batch,
+                     binfo,
+                     offline_batch,
+                     offline_binfo,
+                     weight=alf.nest.get_nest_size(offline_batch, 1) /
+                     mini_batch_size)
+                if do_summary:
+                    if exp:
+                        self.summarize_train(exp, train_info, loss_info,
+                                             params)
+                    # TODO: separate summarization
+                    self.summarize_train(offline_exp, offline_train_info,
```

Comment:
Simply adding `with summary_scope('offline')` should work.

Reply:
Added